### PR TITLE
Add to cart

### PR DESCRIPTION
**Pull request by pearllaw**
issue 3

### CODE REVIEW

**Comment by pearllaw** [comment]
issue 3

File: main.js [hunk 1]
```diff
@@ -144,9 +142,16 @@ function renderCatalog(catalog) {
   return $container
 }
 
-// Click Event Listener
-var $catalogView = document.querySelector('[data-view]')
+// Function rendering cart item count
```
Comment (thebearingedge) [inline main.js]: Please delete comments like this. The function and parameter names should be descriptive enough.

File: main.js [hunk 2]
```diff
@@ -159,6 +164,18 @@ $catalogView.addEventListener('click', function (event) {
...
 })
 
+// Click Event Listener on details page
+var $detailView = document.querySelector('[data-view="details"]')
+$detailView.addEventListener('click', function (event) {
+  if (event.target.getAttribute('id') === 'button') {
+    var item = app.details.item
+    app.cart.push(item)
+    app.view = 'cart'
+    renderCartCount(app.cart)
```
Comment (thebearingedge) [inline main.js]: You are not doing anything with `renderCartCount`'s return value here.

File: main.js [hunk 3]
```diff
@@ -227,10 +249,12 @@ function renderApp(app) {
   var $view = document.querySelector('[data-view="' + app.view + '"]')
   if (app.view === 'catalog') {
     $view.innerHTML = ''
+    $view.appendChild(renderCartCount(app.cart))
```
Comment (thebearingedge) [inline main.js]: Put a separate element in your HTML file to hold the rendered cart count.

File: main.js [hunk 4]
```diff
@@ -227,10 +249,12 @@ function renderApp(app) {
...
     $view.appendChild(renderCatalog(app.catalog.items))
   }
-  if (app.view === 'details') {
+  if (app.view === 'details' || app.view === 'cart') {
```
Comment (thebearingedge) [inline main.js]: The cart view is not until a later feature (view cart summary). The cart count should be visible at all times in the upper corner of the screen.

**Review by thebearingedge — CHANGES_REQUESTED** [review]
Ok, you're on your way. A couple things. 
1. Switch back to this branch and make the requested changes.
2. Please fix your layout of the item details, the image is too large. Try to make your layout look like the wireframe.
3. It's not a good idea to work ahead unless your open Pull Request is _perfect_. Otherwise you will end up with merge conflicts.

This is your current branching history:
screenshot 2018-09-10 08 56 51

This is our goal:
screenshot 2018-09-10 08 57 26

**Comment by thebearingedge** [comment]
Please reopen when ready.

**Comment by pearllaw** [comment]
issue 3

**Review by thebearingedge — APPROVED** [review]

**Pull request merged**